Document/0x04f-Testing-Network-Communication.md

```diff
@@ -200,30 +200,115 @@ When both the client and server are controlled by the same organization and used
+- `DH_anon`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `DHE_RSA`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `DHE_DSS`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `ECDHE_ECDSA`[rfc8422](https://tools.ietf.org/html/rfc8422)
+- `ECDHE_PSK` [rfc8422](https://tools.ietf.org/html/rfc8422) [rfc5489](https://tools.ietf.org/html/rfc5489)
+- `ECDHE_RSA` [rfc8422](https://tools.ietf.org/html/rfc8422)
+
+Block Ciphers:
+- `DES` [rfc4772](https://tools.ietf.org/html/rfc4772)
+- `DES_CBC` [rfc1829](https://tools.ietf.org/html/rfc1829)
+- `3DES` [rfc2420](https://tools.ietf.org/html/rfc2420)
+- `3DES_EDE_CBC`[rfc2420](https://tools.ietf.org/html/rfc2420)
+- `AES_128_CBC`[rfc3268](https://tools.ietf.org/html/rfc3268)
+- `AES_128_GCM` [rfc5288](https://tools.ietf.org/html/rfc5288)
+- `AES_256_CBC`[rfc3268](https://tools.ietf.org/html/rfc3268)
+- `AES_256_GCM`[rfc5288](https://tools.ietf.org/html/rfc5288)
+- `RC4_40` [rfc7465](https://tools.ietf.org/html/rfc7465)
+- `RC4_128` [rfc7465](https://tools.ietf.org/html/rfc7465)
+- `CHACHA20_POLY1305 ` [rfc7905](https://tools.ietf.org/html/rfc7905) [rfc7539](https://tools.ietf.org/html/rfc7539)
+
+Integrity Check Algorithms:
+- `MD5` [rfc6151](https://tools.ietf.org/html/rfc6151)
+- `SHA` [rfc6234](https://tools.ietf.org/html/rfc6234)
+- `SHA256` [rfc6234](https://tools.ietf.org/html/rfc6234)
+- `SHA384` [rfc6234](https://tools.ietf.org/html/rfc6234)
+
+**Remark: The efficiency of a cipher suite depends on the efficiency of its algorithms.**
+
+
+**In the following, we’ll present the recommended cipher suites to use with TLS:**
+
+IANA recommended cipher suites [rfc8447](https://tools.ietf.org/html/rfc8447#section-8) 
+[IANA_CIPHERS](https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4):
+
+- `TLS_DHE_RSA_WITH_AES_128_GCM_SHA256`
+- `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384`
+- `TLS_DHE_PSK_WITH_AES_128_GCM_SHA256`
+- `TLS_DHE_PSK_WITH_AES_256_GCM_SHA384`
+- `TLS_AES_128_GCM_SHA256`
+- `TLS_AES_256_GCM_SHA384`
+- `TLS_CHACHA20_POLY1305_SHA256`
+- `TLS_AES_128_CCM_SHA256`
 - `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256`
-- `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`
 - `TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384`
+- `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`
 - `TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384`
-- `TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256`
+- `TLS_DHE_RSA_WITH_AES_128_CCM`
+- `TLS_DHE_RSA_WITH_AES_256_CCM`
+- `TLS_DHE_PSK_WITH_AES_128_CCM`
+- `TLS_DHE_PSK_WITH_AES_256_CCM`
 - `TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256`
+- `TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256`
+- `TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256`
+- `TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256`
+- `TLS_DHE_PSK_WITH_CHACHA20_POLY1305_SHA256`
+- `TLS_ECDHE_PSK_WITH_AES_128_GCM_SHA256`
+- `TLS_ECDHE_PSK_WITH_AES_256_GCM_SHA384`
+- `TLS_ECDHE_PSK_WITH_AES_128_CCM_SHA256`
 
```

Comment:
Note that in https://tools.ietf.org/html/rfc8446#section-9.1 there is at least oen required cipher suite config for tls 1.3. i guess it would be good to include that as well

Reply:
there are  tls 1.3 recommended cipher suites by IANA that i ahad introduced in the list just check again there are the ones with no keyexchange algorithm and the are the same in the reference you talked about